src/rgw/rgw_sal_motr.cc

```diff
@@ -905,25 +905,50 @@ std::unique_ptr<Object> MotrBucket::get_object(const rgw_obj_key& k)
 int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max, ListResults& results, optional_yield y)
 {
   int rc;
+  if (max <= 0)  // Return an emtpy response.
+    return 0;
```

Comment:
Hi @sjain09, as you asked, we can return `-ERR_INVALID_REQUEST` on -ve values for --max-keys as per AWS behavior or we can keep the empty response as seen in rados+radosgw.

Reply:
I think we should return -EINVAL for max < 0 values.

AWS response for max-key=0 is as below - 
```
#aws s3api list-objects --bucket sjain.bucket --profile aws --max-key 0
{
    "IsTruncated": false,
    "Marker": "",
    "Name": "sjain.bucket",
    "Prefix": "",
    "MaxKeys": 0,
    "EncodingType": "url"
}
``` 

For max < 0 , the response is -
```
#aws s3api list-objects --bucket sjain.bucket --profile aws --max-key -1

An error occurred (InvalidArgument) when calling the ListObjects operation: Argument maxKeys must be an integer between 0 and 2147483647
```

Reply:
Hi, so as discussed with @kupranay, we should keep the behavior same as that of ceph, so probably no change will be needed. @andriytk any views??

Reply:
What's the Ceph's behaviour and Amazon behaviour? Is there any S3 spec about it?

Reply:
on -ve values for --max-keys,
ceph => returns response with 0 keys.
aws   => returns InvalidArgument error.
Not aware of any s3 spec than specified by aws.

Reply:
Thanks. I'd suggest returning an error, like AWS does.

Reply:
Okay. @kupranay, @ujjwalpl can you pls suggest the final approach?
@sjain09,  @andriytk, & I think we should return an error in such this case.

Reply:
I think our overall goal for S3 API is to be compatible with AWS as much as possible, not with RADOS-RGW.

Reply:
Hi all, so sorry observed this just now.
```c++
int RGWListBucket::parse_max_keys()
{
  // Bound max value of max-keys to configured value for security
  // Bound min value of max-keys to '0'
  // Some S3 clients explicitly send max-keys=0 to detect if the bucket is
  // empty without listing any items.
  return parse_value_and_bound(max_keys, max, 0,
			g_conf().get_val<uint64_t>("rgw_max_listing_results"),
			default_max);
}
```
In the ceph code, they are bounding all max-keys values from 0 to 1000. 
so we can't define behavior for the limit without editing ceph specific code. 
in this condition is it good to keep the same behavior then right?
@kupranay , @andriytk , @sjain09

Reply:
@sumedhak27 , what is the output for empty bucket.

Reply:
```bash
╰─➤  aws s3api list-objects-v2 --bucket emptybkt

╰─➤  aws s3api list-objects --bucket emptybkt
```
for empty buckets, there is no response body. why did you ask though?

Reply:
> in this condition, it is good to keep the same behavior then right?

I think so, yes.